central/processlisteningonport/datastore/datastore_impl_test.go

```diff
@@ -254,11 +467,24 @@ func (suite *PLOPDataStoreTestSuite) TestPLOPAddMultipleIndicators() {
 
 	// Fetch inserted PLOP back
 	newPlops, err := suite.datastore.GetProcessListeningOnPort(
-		suite.hasWriteCtx, GetOptions{Namespace: &testNamespace})
+		suite.hasWriteCtx, fixtureconsts.Deployment1)
 	suite.NoError(err)
 
 	suite.Len(newPlops, 1)
-	newPlop, exists := newPlops[fixtureconsts.Deployment1]
-	suite.True(exists)
-	suite.Equal(newPlop[0], plopObjects[0])
+	suite.Equal(*newPlops[0], storage.ProcessListeningOnPort{
+		ContainerName: "test_container1",
+		PodId:         fixtureconsts.PodUID1,
+		DeploymentId:  fixtureconsts.Deployment1,
+		ClusterId:     fixtureconsts.Cluster1,
+		Namespace:     testNamespace,
+		Endpoint: &storage.ProcessListeningOnPort_Endpoint{
+			Port:     1234,
+			Protocol: storage.L4Protocol_L4_PROTOCOL_TCP,
+		},
+		Signal: &storage.ProcessSignal{
+			Name:         "test_process1",
+			Args:         "test_arguments1",
+			ExecFilePath: "test_path1",
+		},
+	})
```

Comment:
Unit tests look good. Could you try writing datastore_impl.go without the closed field and without getting existing plops and see if the unit tests pass?